java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrowUtils.java

```diff
@@ -0,0 +1,219 @@
+   */
+  public static org.apache.arrow.vector.types.pojo.Schema avroToArrowSchema(Schema schema) {
+
+    Preconditions.checkNotNull(schema, "Avro Schema object can't be null");
+    List<Field> arrowFields = new ArrayList<>();
+
+    Schema.Type type = schema.getType();
+
+    if (type == Type.RECORD) {
+      throw new NotImplementedException();
+    } else if (type == Type.MAP) {
+      throw new NotImplementedException();
+    } else if (type == Type.UNION) {
+      throw new NotImplementedException();
+    } else if (type == Type.ARRAY) {
+      throw new NotImplementedException();
+    } else if (type == Type.ENUM) {
+      throw new NotImplementedException();
+    } else if (type == Type.NULL) {
+      throw new NotImplementedException();
+    } else {
+      final FieldType fieldType = new FieldType(true, getArrowType(type), null, null);
+      arrowFields.add(new Field("f0", fieldType, null));
+    }
+
+    return new org.apache.arrow.vector.types.pojo.Schema(arrowFields, null);
+  }
+
+  /**
+   * Create consumers to consume avro values from decoder, will reduce boxing/unboxing operations.
+   */
+  public static List<Consumer> createAvroConsumers(VectorSchemaRoot root) {
+    List<Consumer> consumers = new ArrayList<>();
+
+    for (ValueVector vector : root.getFieldVectors()) {
+      Consumer consumer;
+      switch (vector.getMinorType()) {
+        case INT:
+          consumer = new AvroIntConsumer((IntVector) vector);
+          break;
+        case VARBINARY:
+          consumer = new AvroBytesConsumer((VarBinaryVector) vector);
+          break;
+        case VARCHAR:
+          consumer = new AvroStringConsumer((VarCharVector) vector);
+          break;
+        case BIGINT:
+          consumer = new AvroLongConsumer((BigIntVector) vector);
+          break;
+        case FLOAT4:
+          consumer = new AvroFloatConsumer((Float4Vector) vector);
+          break;
+        case FLOAT8:
+          consumer = new AvroDoubleConsumer((Float8Vector) vector);
+          break;
+        case BIT:
+          consumer = new AvroBooleanConsumer((BitVector) vector);
+          break;
+        default:
+          throw new RuntimeException("could not get consumer from type:" + vector.getMinorType());
+      }
+      consumers.add(consumer);
+    }
+    return consumers;
+  }
+
+  /**
+   * Iterate the given Avro {@link Decoder} object to fetch the data and transpose it to populate
+   * the given Arrow Vector objects.
+   * @param decoder avro decoder to read data.
+   * @param root Arrow {@link VectorSchemaRoot} object to populate
+   */
+  public static void avroToArrowVectors(Decoder decoder, VectorSchemaRoot root) {
+
+    Preconditions.checkNotNull(decoder, "Avro decoder object can't be null");
+    Preconditions.checkNotNull(root, "VectorSchemaRoot object can't be null");
+
+    allocateVectors(root, DEFAULT_BUFFER_SIZE);
+
+    List<Consumer> consumers = createAvroConsumers(root);
```

Comment:
it seems like we might need to understand more about decoders and the order they are expected to [decode(https://avro.apache.org/docs/1.8.2/api/java/org/apache/avro/io/ValidatingDecoder.html) in.  This can also be for a follow-up PR